### PR TITLE
service discovery - registering myself and discovering nodes in my cluster

### PR DESCRIPTION
**Pull request by eishay**

### CODE REVIEW

File: shoebox/modules/common/test/com/keepit/common/zookeeper/ServiceDiscoveryLiveTest.scala [hunk 1]
```diff
@@ -28,25 +31,34 @@ class ServiceDiscoveryLiveTest extends Specification with TestInjector {
...
-          println("new node: " + zk.createNode(Node(s"$basePath/services/TEST_MODE/TEST_MODE_"), null, EPHEMERAL_SEQUENTIAL))
-          println("sleeping")
-          println(zk.getChildren(Path(s"$basePath/services/TEST_MODE")) mkString ",")
-          zk.getChildren(Path(s"$basePath/services/TEST_MODE")).size === 3
+          val thirdNode = zk.createNode(Node("/fortytwo/services/TEST_MODE/TEST_MODE_"), null, EPHEMERAL_SEQUENTIAL)
+          zk.set(thirdNode, Json.toJson(amazonInstanceInfo.copy(localHostname = "third")).toString)
+          println("new node: " + thirdNode, null, EPHEMERAL_SEQUENTIAL)
+          println("sleeping 1")
+          Thread.sleep(10000)
+          println(zk.getChildren(Path("/fortytwo/services/TEST_MODE")) mkString ",")
```
Comment (gmethvin) [inline shoebox/modules/common/test/com/keepit/common/zookeeper/ServiceDiscoveryLiveTest.scala]: why println instead of logs?

Comment (eishay) [inline shoebox/modules/common/test/com/keepit/common/zookeeper/ServiceDiscoveryLiveTest.scala]: leftovers from testing locally, will move to log

File: shoebox/modules/common/test/com/keepit/common/zookeeper/ServiceDiscoveryLiveTest.scala [hunk 2]
```diff
@@ -28,25 +31,34 @@ class ServiceDiscoveryLiveTest extends Specification with TestInjector {
...
+          zk.getChildren(Path("/fortytwo/services/TEST_MODE")).size === 3
           discovery.isLeader() === false
+          discovery.myClusterSize === 2
+          println("sleeping 2 - about to delete")
+          Thread.sleep(10000)
```
Comment (gmethvin) [inline shoebox/modules/common/test/com/keepit/common/zookeeper/ServiceDiscoveryLiveTest.scala]: This seems like a long time to sleep in a test. Can we use a fake clock or a maybe a smaller time interval?

Comment (eishay) [inline shoebox/modules/common/test/com/keepit/common/zookeeper/ServiceDiscoveryLiveTest.scala]: True. The test is not running by default (skipAll). Its running against a live server so its depending on external IO. I'll add more tests to the ServiceDiscoveryTest (not the ServiceDiscoveryLiveTest) that will involve a fake ZK client and no sleeps.

**Pull request closed without merging**